__tests__/integration/helpers/wallet.helper.js

```diff
@@ -274,82 +274,94 @@ export function waitForWalletReady(hWallet) {
 }
 
 /**
- * Translates the tx success event into a promise.
- * Waits for the wallet event that indicates this transaction id has been fully integrated
- * into the lib's local caches. Actions that depend on this state can be executed after the
- * successful response from this function.
+ * Waits for the wallet to receive the transaction from a websocket message
+ * and process the history
+ *
  * @param {HathorWallet} hWallet
  * @param {string} txId
  * @param {number} [timeout] Timeout in milisseconds. Default value defined on test-constants.
- * @returns {Promise<SendTxResponse>}
+ * @returns {Promise<IHistoryTx>}
  */
 export async function waitForTxReceived(hWallet, txId, timeout) {
   const startTime = Date.now().valueOf();
-  let alreadyResponded = false;
-  const existingTx = await hWallet.getTx(txId);
+  let timeoutHandler;
+  let timeoutReached = false;
+  const timeoutPeriod = timeout || TX_TIMEOUT_DEFAULT;
+  // Timeout handler
+  timeoutHandler = setTimeout(() => {
+    timeoutReached = true;
+  }, timeoutPeriod);
 
-  // If the transaction was already received, return it immediately
-  if (existingTx) {
-    return existingTx;
-  }
+  let storageTx = await hWallet.getTx(txId);
 
-  // Only return the positive response after the transaction was received by the websocket
-  return new Promise(async (resolve, reject) => {
-    // Event listener
-    const handleNewTx = newTx => {
-      // Ignore this event if we didn't receive the transaction we expected.
-      if (newTx.tx_id !== txId) {
-        return;
-      }
+  // We consider that the tx was received after it's in the storage
+  // and the history processing is finished
+  while (!storageTx || storageTx.processingStatus !== TxHistoryProcessingStatus.FINISHED) {
+    if (timeoutReached) {
+      break;
+    }
 
-      // This is the correct transaction: resolving the promise.
-      resolveWithSuccess(newTx);
-    };
-    hWallet.on('new-tx', handleNewTx);
+    // Tx not found, wait 1s before trying again
+    await delay(1000);
+    storageTx = await hWallet.getTx(txId);
+  }
 
-    // Timeout handler
-    const timeoutPeriod = timeout || TX_TIMEOUT_DEFAULT;
-    setTimeout(async () => {
-      hWallet.removeListener('new-tx', handleNewTx);
+  // Clean timeout handler
+  if (timeoutHandler) {
+    clearTimeout(timeoutHandler);
+  }
 
-      // No need to respond if the event listener worked.
-      if (alreadyResponded) {
-        return;
-      }
+  if (timeoutReached) {
+    // Throw error in case of timeout
+    throw new Error(`Timeout of ${timeoutPeriod}ms without receiving the tx with id ${txId}`);
+  } else {
```

Comment:
We don't need this `else`. From line 316 on on we're sure `timeoutReached` is false

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/590/commits/5c1085e54005ee64543b98b8f0a8d69358e6cc4b